### ISSUE TITLE
feat: add hive 

### ISSUE THREAD

**Issue by Gorniv** [Baseflow/flutter_cache_manager#121]
Use hive(https://github.com/hivedb/hive)  instead SQLite.

**Comment by petermichaux** [comment]
@Gorniv Why use hive instead of SQLite?

**Comment by Nisthar** [comment]
@petermichaux because its much faster than sqlite.

**Comment by Gorniv** [comment]
cache manager uses SQLite only like key-value storage, but hive faster! And I have crashed at the android device with SQLite error.

**Comment by Nisthar** [comment]
also i think hive have a simple api than sqlite. so it would be easy to implement.

**Comment by ahmedkassem** [comment]
same problem here, sqlite causes this problem 
Warning database has been locked for 0:00:10.000000. Make sure you always use the transaction object for database operations during a transaction

**Comment by Gorniv** [comment]
hivedb support web!

**Comment by britannio** [comment]
This package is used to store files, something that cannot be done on web as `dart:io` isn't available so web support isn't a concern.

Maybe migrating to [moor](https://pub.dev/packages/moor) is viable? A dev version using [moor_ffi ](https://pub.dev/packages/moor_ffi)would be awesome

**Comment by Gorniv** [comment]
hivedb work in web. Read docs of hivedb

**Comment by britannio** [comment]
> hivedb work in web. Read docs of hivedb

I know, read my message again.

**Comment by Gorniv** [comment]
sorry, I think you write about hive)

**Comment by venkatd** [comment]
https://github.com/Baseflow/flutter_cache_manager/blob/ed5a205bcb5e816d05113c0360fea02aadc7e2be/flutter_cache_manager/lib/src/storage/cache_info_repository.dart

If someone is interested in adding support, they would need to implement the CacheInfoRepository interface:
https://github.com/Baseflow/flutter_cache_manager/blob/ed5a205bcb5e816d05113c0360fea02aadc7e2be/flutter_cache_manager/lib/src/storage/cache_info_repository.dart

**Comment by joe-getcouragenow** [comment]
This one works well for me:

https://github.com/vishna/flutter_cache_manager_hive

**Comment by a0v0** [comment]
any update?

**Comment by yangsfang** [comment]
I don't think this is a Hive vs sqflite debate. `flutter_cache_manager` is a library, and it should have flexibility depending on what the developer needs to use in the rest of the project. The worst case is to dictate sqflite or hive when the developer is using the other one, and that means pulling in two dependencies, increases compile time, increases app size, increases app loading time, increases memory jank, increases garbage collection interruption, giving a worse user experience. 

We don't want that. We want to have the flexibility, which is why interface pattern is so cherished by developers. We do have the [cache_info_repository.dart](https://github.com/Baseflow/flutter_cache_manager/blob/develop/flutter_cache_manager/lib/src/storage/cache_info_repositories/cache_info_repository.dart) like an interface, maybe this file should be pulled into a separate Dart package on its own, and refactor `flutter_cache_manager` into a sqflite implementation. This allows someone to make the Hive implementation. Then we are all good to choose whichever dependency we like.

**Comment by OnClickListener2048** [comment]
> hivedb work in web. Read docs of hivedb

why not drift

**Comment by OnClickListener2048** [comment]
> This package is used to store files, something that cannot be done on web as `dart:io` isn't available so web support isn't a concern.
> 
> Maybe migrating to [moor](https://pub.dev/packages/moor) is viable? A dev version using [moor_ffi ](https://pub.dev/packages/moor_ffi)would be awesome

why not drift
